app/controllers/incomeSources/add/AddBusinessNameController.scala

```diff
@@ -70,12 +70,15 @@ class AddBusinessNameController @Inject()(authenticate: AuthenticationPredicate,
 
   private lazy val journeyType: JourneyType = JourneyType(Add, SelfEmployment)
 
-  private def getBusinessName(isChange: Boolean)
+  private def getBusinessName(isChange: Boolean, sessionData: Option[UIJourneySessionData])
                              (implicit user: MtdItUser[_]): Future[Option[String]] = {
     if (isChange)
-      sessionService.getMongoKeyTyped[String](AddIncomeSourceData.businessNameField, journeyType).flatMap {
-        case Right(nameOpt) => Future.successful(nameOpt)
-        case Left(ex) => Future.failed(ex)
+      sessionData match {
+        case Some(session) => session.addIncomeSourceData match {
+          case Some(data) => Future.successful(data.businessName)
+          case None => Future.failed(new Error("No session data found"))
+        }
```

Comment:
The checks for this have already been executed in the utils file. Throwing an additional error here could result in loss of stack trace. It's not necessary to generate errors throughout every level of the application. If there is a need to throw something, perhaps consider using an exception rather than an error (errors should only be used for JVM/fatal issues).
https://www.baeldung.com/scala/nonfatal-vs-exception#:~:text=Error%20represents%20fatal%20issues%20that,be%20handled%20or%20recovered%20from.